### PR TITLE
Fixed deleting ifroute when an interface is deleted

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: src/lib/y2network/config.rb [hunk 1]
```diff
@@ -101,9 +101,14 @@ def initialize(interfaces: InterfacesCollection.new, connections: [], routing: R
...
+    # Writes only changes agains original configuration if the original configuration
+    # is provided
+    #
+    # @param original [Y2Network::Config] configuration used for detecting changes
+    #
     # @see Y2Network::ConfigWriter
-    def write
-      Y2Network::ConfigWriter.for(source).write(self)
+    def write(original: nil)
+      Y2Network::ConfigWriter.for(source).write(self, old_config = original)
```
Comment (jreidinger) [inline src/lib/y2network/config.rb]: is assignment to old_config really needed?

File: src/lib/y2network/sysconfig/config_writer.rb [hunk 2]
```diff
@@ -37,31 +37,47 @@ def write(config, old_config = nil)
...
           file.routes = routes
           file.save
         end
 
-        # update /etc/sysconfig/network/routes file
-        file = routes_file_for(nil)
-        file.routes = find_routes_for(nil, config.routing.routes)
-        file.save
+        # Actions needed for removed interfaces
+        removed_ifaces = old_config ? old_config.interfaces.to_a - config.interfaces.to_a : []
```
Comment (jreidinger) [inline src/lib/y2network/sysconfig/config_writer.rb]: so old config can be nil? If so, please document in method documentation.

File: test/y2network/sysconfig/config_writer_test.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@
         source:     :sysconfig
       )
     end
-    let(:old_config) { instance_double(Y2Network::Config, dns: double("dns")) }
+    let(:old_config) { instance_double(Y2Network::Config, dns: double("dns"), interfaces: nil) }
```
Comment (jreidinger) [inline test/y2network/sysconfig/config_writer_test.rb]: I expect here more code that tests also that removal of files and other parts that is added.

BTW is only network-ng affected or also master?

Comment (mchf) [reply test/y2network/sysconfig/config_writer_test.rb]: should be network-ng specific, but will check in master too.

**Pull request merged**